src/domain/users/user-organizations.repository.ts

```diff
@@ -0,0 +1,305 @@
+      where,
+      relations,
+    });
+  }
+
+  public async inviteUsers(args: {
+    authPayload: AuthPayload;
+    orgId: Organization['id'];
+    users: Array<{
+      address: `0x${string}`;
+      role: UserOrganizationRole;
+    }>;
+  }): Promise<
+    Array<{
+      userId: User['id'];
+      orgId: Organization['id'];
+      role: keyof typeof UserOrganizationRole;
+      status: keyof typeof UserOrganizationStatus;
+    }>
+  > {
+    const signer = await this.getOrgAndSignerOrFail(args);
+
+    this.assertUserOrgIsActive(signer.userOrg);
+
+    const invitations: Array<{
+      userId: User['id'];
+      orgId: Organization['id'];
+      role: keyof typeof UserOrganizationRole;
+      status: keyof typeof UserOrganizationStatus;
+    }> = [];
+
+    await this.postgresDatabaseService.transaction(async (entityManager) => {
+      for (const userToInvite of args.users) {
+        const invitedUser = await this.walletsRepository.findOneByAddress(
+          userToInvite.address,
+          { user: true },
+        );
+
+        let invitedUserId = invitedUser?.id;
+
+        if (!invitedUserId) {
+          invitedUserId = await this.usersRepository.create(
+            UserStatus.PENDING,
+            entityManager,
+          );
+
+          await this.walletsRepository.create(
+            {
+              walletAddress: userToInvite.address,
+              userId: invitedUserId,
+            },
+            entityManager,
+          );
+        }
+
+        await entityManager.insert(DbUserOrganization, {
+          user: { id: invitedUserId },
+          organization: signer.org,
+          role: userToInvite.role,
+          status: UserOrganizationStatus.INVITED,
+        });
+
+        invitations.push({
+          userId: invitedUserId,
+          orgId: signer.org.id,
+          role: getEnumKey(UserOrganizationRole, userToInvite.role),
+          status: getEnumKey(
+            UserOrganizationStatus,
+            UserOrganizationStatus.INVITED,
+          ),
+        });
+      }
+    });
+
+    return invitations;
+  }
+
+  public async updateStatus(args: {
+    authPayload: AuthPayload;
+    orgId: Organization['id'];
+    status: UserOrganization['status'];
+  }): Promise<void> {
+    const signer = await this.getOrgAndSignerOrFail(args);
+
+    if (signer.userOrg.status !== UserOrganizationStatus.INVITED) {
+      throw new ConflictException('Invite is not pending.');
+    }
+
+    await this.postgresDatabaseService.transaction(async (entityManager) => {
+      await entityManager.update(DbUserOrganization, signer.userOrg.id, {
+        status: args.status,
+      });
+
+      const isActivatingUserOrg = args.status === UserOrganizationStatus.ACTIVE;
+      const isUserPending = signer.user.status === UserStatus.PENDING;
+      if (isActivatingUserOrg && isUserPending) {
+        await entityManager.update(DbUser, signer.user.id, {
+          status: UserStatus.ACTIVE,
+        });
+      }
+    });
+  }
```

Comment:
We should split this function: one for activation and one for declining. The latter doesn't need to be a transaction. The user update should be a user repository method as well.